python/ray/actor.py

```diff
@@ -36,7 +35,6 @@
 
 
 @PublicAPI
-@client_mode_hook(auto_init=False)
```

Comment:
cc @ijrsvt It's safe to drop this if the client mode version of this method is identical right? (see deleted function in api.py below)

Reply:
OH hmm, I'm really not sure what the large comment in `api.py` is describing :(

Reply:
@ericl do you remember what's going on there and if it's safe to remove this?
This stuff was introduced in a PR https://github.com/ray-project/ray/pull/13306 you reviewed long, long ago.

Reply:
I realized that we probably need to keep this method in `api.py`, in case anyone still using `from ray.util.client import ray` to annotate their methods upgrades their ray version. I'll duplicate the changes to there for now, can figure out if its safe to refactor in a different PR.